### PR TITLE
ZEPPELIN-4081. when the python process is killed,the task state is still running

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
This PR will break python code execution if the python process is existed. Besides that, I also improve the error message for ipython interpreter although it doesn't have such issue.

### What type of PR is it?
[Bug Fix]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-4081

### How should this be tested?
* Unit test is added

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by AyWa** [comment]
For Ipython, there is a potential issue where the ipython server does not die, but the ipython kernel does.
I just open a PR: https://github.com/apache/zeppelin/pull/3339
it is not ready yet because need to do some cleanup, but wanted to bring it because those 2 pr could be related.

**Comment by zjffdu** [comment]
@felixcheung I have added it in LICENCE. @AyWa Thanks for your caring on ipython.

File: python/src/test/java/org/apache/zeppelin/python/IPythonInterpreterTest.java [hunk 1]
```diff
@@ -235,4 +238,30 @@ public void testGrpcFrameSize() throws InterpreterException, IOException {
...
+  @Test
+  public void testIPythonProcessKilled() throws InterruptedException, TimeoutException {
+    final Waiter waiter = new Waiter();
+    Thread thread = new Thread() {
+      @Override
+      public void run() {
+        try {
+          InterpreterResult result = interpreter.interpret("import time\ntime.sleep(1000)",
+                  getInterpreterContext());
+          waiter.assertEquals(InterpreterResult.Code.ERROR, result.code());
```
Comment (zjffdu) [inline python/src/test/java/org/apache/zeppelin/python/IPythonInterpreterTest.java]: Junit can only catch assertion fail in main thread, so here use concurrentunit to do unit test in multiple thread

**Comment by zjffdu** [comment]
Will merge if no more comments

**Comment by zjffdu** [comment]
Sorry @AyWa  I just reverted this PR, because I found it may cause travis CI failed.

**Comment by AyWa** [comment]
Hey,
you mean https://github.com/apache/zeppelin/pull/3339 ? No worries should I looks to something ?

**Comment by zjffdu** [comment]
@AyWa Sorry again, I merge your PR again, because I found the Ci fail is due to JVM memory setting. I have fixed it in https://github.com/apache/zeppelin/commit/1a8667c655bb848fd2d07e1fe94092190454a5dd

**Pull request closed without merging**